c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -1709,24 +1713,20 @@ class TranslateASTVisitor final
     }
 
   private:
-    bool warnOnFlexibleArrayDecl(FieldDecl *D) {
+    bool maybeFlexibleArrayDecl(FieldDecl *D) {
         const ASTRecordLayout &Layout =
             Context->getASTRecordLayout(D->getParent());
         unsigned FieldCount = Layout.getFieldCount();
 
         if (auto CA = dyn_cast_or_null<ConstantArrayType>(
                 D->getType().getTypePtr())) {
-            // If the array has a size of 1, and struct field count is
-            // greater than 1, and if the (struct field count - 1) is equal to
-            // the index, it is most likely a flexible array.
-            if (CA->getSize() == 1 && FieldCount > 1 &&
-                FieldCount - 1 == D->getFieldIndex() &&
-                !warnedFlexibleArrayDecls.count(D)) {
-                // Insert the Decl into the set, if it has not been warned about
-                // yet.
-                warnedFlexibleArrayDecls.insert(D);
-                return true;
-            }
+            // If the array has a size of 1, struct field count is greater
```

Comment:
Can't the array size be 0 as well?

Reply:
Yes, that is a GNU extension. We don't need to warn about that because people don't need to fix their code, though. We handle it at https://github.com/immunant/c2rust/pull/68/files#diff-4c1124a9b33d10318140e1ae99bd3189R138

I actually want to get rid of the warning in AstExporter entirely, but am waiting on a response from @perl on the bug before I do.